test/wallet-test.js

```diff
@@ -2279,4 +2299,131 @@ describe('Wallet', function() {
       assert.equal(confirmedCount, 1);
     });
   });
+
+  describe('Wallet Name Claims', function() {
+    let wallet, update;
+    const network = Network.get('regtest');
+    const workers = new WorkerPool({enabled: false});
+    const wdb = new WalletDB({network, workers});
+    const lockup = 6800503496047;
+    const name = 'cloudflare';
+    const nameHash = rules.hashString(name);
+
+    before(async () => {
+      await wdb.open();
+      wallet = await wdb.create();
+
+      for (let i = 0; i < 3; i++) {
+        const entry = nextEntry(wdb);
+        await wdb.addBlock(entry, []);
+      }
+
+      // Use a fresh wallet.
+      const bal = await wallet.getBalance();
+      assert.equal(bal.tx, 0);
+      assert.equal(bal.coin, 0);
+      assert.equal(bal.unconfirmed, 0);
+      assert.equal(bal.confirmed, 0);
+      assert.equal(bal.ulocked, 0);
+      assert.equal(bal.clocked, 0);
+    });
+
+    after(async () => {
+      await wdb.close();
+    });
+
+    it('should not have any cloudflare state', async () => {
+      const nameinfo = await wallet.getNameState(nameHash);
+      assert.deepEqual(nameinfo, null);
+    });
+
+    it('should confirm cloudflare CLAIM', async () => {
+      const claim = await wallet.sendFakeClaim('cloudflare');
+      assert(claim);
+
+      const tx = claim.toTX(network, wdb.state.height + 1);
+      const entry = nextEntry(wdb);
+      await wdb.addBlock(entry, [tx]);
+
+      const ns = await wallet.getNameState(nameHash);
+      const json = ns.getJSON(wdb.state.height, network);
+      assert.equal(json.name, 'cloudflare');
+      assert.equal(json.state, 'LOCKED');
+
+      const bal = await wallet.getBalance();
+      assert.equal(bal.tx, 1);
+      assert.equal(bal.coin, 1);
+      assert.equal(bal.unconfirmed, lockup);
+      assert.equal(bal.confirmed, lockup);
+      assert.equal(bal.ulocked, lockup);
+      assert.equal(bal.clocked, lockup);
```

Comment:
For the sake of being totally thorough, I wonder if the balance should be checked before the block is added to assert that the unconfirmed / confirmed values are computed properly through the process.

Reply:
Assuming that no additional tests are added in between this `it` block and the `before` block, this is already being tested. See:
https://github.com/handshake-org/hsd/pull/438/files#diff-ccdf9d4d9aaae8ebafdd75a1f30a1b4dR2322-R2328

I do not like that assumption, so I will add in the assertions to this particular `it` block.